drivers/telemetry/telemetry_intel.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _TELEMETRY_INTEL_
+#define _TELEMETRY_INTEL_
```

Comment:
@nashif @MaureenHelm @teburd This general concept of adding hw/vendor specific telemetry interface requires feedback. I don't see any existing telemetry infra in Zephyr now, so starting with a vendor specific interface seems pragmatic start, but requires attention anyways. FYI @mwasko  @marcinszkudlinski

Reply:
Hi, what I've put here is only a telemetry base structure, to show the purpose of this PR and that it works. This structure cannot be modified anyway, as it is supposed to be seen by host like it is. I've tried to make the last two commits a separate PR, so we can discuss it there, but there is one dependency that cannot be overcome until this PR is merged and it has to go together. So my suggestion is to postpone this this discussion till next PR's with proper API's will be ready - the ones that actually fill the data.

Reply:
@kv2019i as you metioned, the telemetry concept is unique for Intel configuration for now, so I recommend to continue with integration of current solution as it is in the first step. There is no telemetry interface that we could align with today.

@nashif, @MaureenHelm, @teburd it would be probably ask to you to prepare generic Telemetry interface proposition using Intel implementation as an input. Once the generic interface definition will be ready we can apply it to Intel telemetry driver.